tests_e2e/tests/scripts/agent_ext_policy-verify_uninstall_success.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env pypy3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Verify if the agent reported update status to CRP via status file
+#
+import argparse
+import glob
+import json
+
+from assertpy import fail
+
+from tests_e2e.tests.lib.logging import log
+
+
+def extension_found_in_agent_status_file(ext_name: str) -> bool:
+    # Check if the provided extension name is present in the agent status file, under handlerAggregateStatus.
+    # If the name is not present, the uninstall operation was successful.
+    agent_status_file = "/var/lib/waagent/history/*/waagent_status.json"
```

Comment:
this is not a reliable way to check the status, since there is a new file per goal state and there may be new goal states while the test is running. to make it reliable you would also need to verify that the report actually corresponds to the  goal state you are checking (you could use the activity id)

checking waagent.log may be a simpler option

Reply:
can be on next pr

Reply:
Marked as TODO